File: melosys-eessi-app/src/test/java/no/nav/melosys/eessi/service/sed/mapper/til_sed/lovvalg/A011MapperTest.java
package no.nav.melosys.eessi.service.sed.mapper.til_sed.lovvalg;

import java.io.IOException;
import java.net.URL;

import com.fasterxml.jackson.databind.ObjectMapper;
import no.nav.melosys.eessi.models.SedType;
import no.nav.melosys.eessi.models.sed.SED;
import no.nav.melosys.eessi.models.sed.medlemskap.impl.MedlemskapA011;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.assertj.core.api.Assertions.assertThat;

class A011MapperTest {

    private final A011Mapper a011Mapper = new A011Mapper();

    private SED a001;

    @BeforeEach
    public void setup() throws IOException {
        URL jsonUrl = getClass().getClassLoader().getResource("mock/sedA001.json");
        a001 = new ObjectMapper().readValue(jsonUrl, SED.class);
    }

    @Test
    void mapFraEksisterendeSedA001() {
        SED a011 = a011Mapper.mapFraSed(a001);

        assertThat(a011.getSedType()).isEqualToIgnoringCase(SedType.A011.toString());
        assertThat(a011.getNav().getBruker().getPerson().getFornavn()).isEqualToIgnoringCase("Testfornavn1");
        assertThat(a011.getMedlemskap()).isInstanceOf(MedlemskapA011.class);
        assertThat(a011.getMedlemskap()).isNotNull();
    }
}

File: melosys-eessi-app/src/test/java/no/nav/melosys/eessi/service/sed/mapper/til_sed/lovvalg/A011MapperTest.kt
package no.nav.melosys.eessi.service.sed.mapper.til_sed.lovvalg

import io.kotest.matchers.nulls.shouldNotBeNull
import io.kotest.matchers.shouldBe
import io.kotest.matchers.types.shouldBeInstanceOf
import no.nav.melosys.eessi.models.SedType
import no.nav.melosys.eessi.models.sed.SED
import no.nav.melosys.eessi.models.sed.medlemskap.impl.MedlemskapA011
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import tools.jackson.databind.json.JsonMapper
import java.io.IOException

class A011MapperTest {

    private val a011Mapper = A011Mapper()
    private lateinit var a001: SED
    private val jsonMapper = JsonMapper.builder().build()

    @BeforeEach
    fun setup() {
        val inputStream = this::class.java.classLoader.getResourceAsStream("mock/sedA001.json")
            ?: throw IOException("File not found: mock/sedA001.json")
        a001 = jsonMapper.readValue(inputStream, SED::class.java)
    }

    @Test
    fun `map fra eksisterende SED A001 forvent korrekt SED A011`() {
        val a011 = a011Mapper.mapFraSed(a001)

        a011.shouldNotBeNull().run {
            sedType shouldBe SedType.A011.toString()

            nav.shouldNotBeNull()
                .bruker.shouldNotBeNull()
                .person.shouldNotBeNull()
                .fornavn shouldBe "Testfornavn1"

            medlemskap.shouldNotBeNull().shouldBeInstanceOf<MedlemskapA011>()
        }
    }
}
